howso/train.amlg

```diff
@@ -705,11 +705,52 @@
 				batch_size 0
 				input_case_index 0
 				output_case_ids []
+
+				;threshold-related variables
+				thresholds_enabled
+					(or
+						(size !autoAblationAbsThresholdMap)
+						(size !autoAblationDeltaThresholdMap)
+						(size !autoAblationRelThresholdMap)
+					)
+				prev_prediction_stats_map {}
+				new_prediction_stats_map {}
+				thresholds_satisfied (false)
 			))
 
 			;split by batches of cases until next analyze
 			(while (< input_case_index (size cases))
 
+				(if (and thresholds_enabled !autoAblationEnabled)
```

Comment:
!autoAblationEnabled or (not skip_ablation) ? ie, I don't think should this block be run during the first training batch before any cases have even been trained, or when only a few cases have been trained.